src/transfer.ts

```diff
@@ -70,6 +70,7 @@ export const transferStatusResponseSchema = z.object(
     fiatAccountId: fiatAccountIdSchema,
     transferId: z.string(),
     transferAddress: z.string(),
+    txHash: z.string().optional(),
```

Comment:
you could make this required in the cases where it's required as follows:
- make the schema union of two schemas, maybe "preTxTransferStatusResponseSchema" and "postTxTransferStatusResponseSchema"
- restrict the transfer statuses allowable in each to the pre-tx statuses and post-tx statuses respectively
- make txHash required in postTx and omit it from preTx

Reply:
Yeah, I had considered this, though there's somewhat of an inconsistency in the types repo re: enforcing these more nuanced semantics (e.g., we don't enforce that fiat account schemas in quote responses match the fiat account type the quote response claims they do), so I wasn't certain if it was worth enforcing this. On top of that, this approach has the quality of being non-breaking, which was incidentally kind of nice. 

That said, I'll update the schemas to enforce these semantics; it's not particularly difficult.

Reply:
ok, it wasn't particularly straightforward, either :sweat_smile: